crm_compassion/models/event_compassion.py

```diff
@@ -482,17 +483,16 @@ def _get_calendar_vals(self):
         :return: dictionary of calendar.event values
         """
         self.ensure_one()
-        number_of_days = 1
-        start_date = fields.Datetime.from_string(self.start_date)
-        if self.end_date:
-            end_date = fields.Datetime.from_string(self.end_date)
-            number_of_days = (end_date - start_date).days
+        time_delta = (fields.Datetime.from_string(self.end_date) -
+                      fields.Datetime.from_string(self.start_date))
+        duration_in_hours = math.ceil(time_delta.days * 24 +
+                                      time_delta.seconds / 3600.0)
         calendar_vals = {
             'name': self.name,
             'compassion_event_id': self.id,
             'categ_ids': [
                 (6, 0, [self.env.ref('crm_compassion.calendar_event').id])],
-            'duration': number_of_days * 8,
+            'duration': duration_in_hours,
```

Comment:
I think now we still have the problem that if start_date == end_date, the duration will be 0? Maybe we should put here `duration_in_hours or 3` to put a minimum of 3 hours duration for instance?